### DISCUSSION TITLE
multivariate DLM

### DISCUSSION THREAD

**Discussion by maria-kuruvilla** [atsa-es/fish550-2023#42] [Q&A]
If you had n timeseries and m covariates for each of the timeseries (including intercept), what would the dimensions of your Q matrix be? Would it be an array with dimensions n x m x m or a matrix with dimensions n*m x n*m?

**Comment by mdscheuerell** [comment]
Hi @maria-kuruvilla,

Q is always a matrix and its dimensions must equal the number of rows in the state/process vector (x_t). So, for your case above, it would indeed be _nm_ x _nm_.

Note, too, that care should be given when setting up the model so that the correct intercepts & slopes apply to the correct response. You may also want to set up a custom Q where all of the intercepts get the same variance and all of the slopes get the same variance, and possibly different from (same as) the intercepts.
